game-app/game-core/src/main/java/games/strategy/triplea/attachments/RulesAttachment.java

```diff
@@ -1064,6 +1104,18 @@ private boolean checkTechs(final GamePlayer player, final TechnologyFrontier tec
     return found >= techCount;
   }
 
+  @VisibleForTesting
+  public boolean checkHasResource(final List<GamePlayer> players) {
```

Comment:
Maybe add JavaDoc as it is (at least for me) hard to read and understand what this method is supposed to be doing.
It seem that the purpose it to check whether the sum of the quantities of all resources from all provided players is smaller than `hasResource[0]`.
Why is does `hasResource` need to be a member attribute of this class? It seems I am missing something, but then also it seems to be not clear from the code.